sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/orc/OrcPartitionReaderFactory.scala

```diff
@@ -80,10 +80,10 @@ case class OrcPartitionReaderFactory(
           isCaseSensitive, dataSchema, readDataSchema, reader, conf)
       }
 
-    if (requestedColIdsOrEmptyFile.isEmpty) {
+    if (requestedColIdsOrEmptyFile._1.isEmpty) {
```

Comment:
ditt.

Please try the following at line 77.
```scala
- val requestedColIdsOrEmptyFile =
+ val (requestedColIdsOrEmptyFile, _) =
```

Reply:
done